spec/database_cleaner/base_spec.rb

```diff
@@ -223,31 +90,19 @@ module DatabaseCleaner
     end
 
     describe "clean_with" do
-      let (:strategy) { double("strategy", clean: true) }
-
-      before do
-        allow(subject).to receive(:create_strategy).with(anything).and_return(strategy)
-      end
-    end
-
-    describe "clean_with" do
-      # FIXME hacky null strategy
-      # because you can't pass a NullStrategy to #clean_with
+      subject(:active_record_cleaner) { Base.new(:active_record) }
 
-      let(:strategy) { double(clean: true) }
-
-      let(:strategy_class) do
-        require "database_cleaner/active_record/truncation"
-        DatabaseCleaner::ActiveRecord::Truncation
-      end
+      let(:strategy_class) { Class.new }
+      let(:strategy) { double }
+      before { allow(strategy_class).to receive(:new).and_return(strategy) }
 
       before do
-        allow(::ActiveRecord::Base).to receive(:connection).and_return(double.as_null_object)
-        allow(strategy_class).to receive(:new).and_return(strategy)
+        stub_const "DatabaseCleaner::ActiveRecord::Truncation", strategy_class
       end
 
       it "should pass all arguments to strategy initializer" do
         expect(strategy_class).to receive(:new).with(:dollar, :amet, ipsum: "random").and_return(strategy)
+        expect(strategy).to receive(:clean)
         subject.clean_with :truncation, :dollar, :amet, ipsum: "random"
```

Comment:
I'd probably use the named subject here for example.

Reply:
Named subject is a brand new concept to me, so I very much value your input, here. Thank you!

I've already named the subject `:active_record_cleaner` in the `describe "clean_with"` context, with the idea that it aids in debugging via better failure messages. But you're saying that rather than invoking that named subject as `subject`, it should be invoked as `active_record_cleaner`? And now I'm realizing that named subject is primarily a readability win? Does this sound right?

Reply:
Yep it should be invoked as its name, subject is a left over from when we recommended one line specs, it was never intended to be invoked by its name, but then it became a symbol of the "subject under test" when really its clearer if this is named to what it is, named subjects are a good cross as it highlights which helper is the subject under test, but encourages better naming.

Reply:
I dig it, man. Thank you for the explanation! I'll update the specs with this new understanding.

Reply:
One last question about this. If it were you, would you remove all invocations of `subject` in it blocks? i.e. the only reason to use `subject` in your specs is to set up a named subject? Basically a version of `let` that carries an "primary SUT" connotation?

I'm not sure where to draw the line, right now. I'm waffling between the path described above, and using bare `subject` when all we need is an implicit `described_class.new`, and named subjects otherwise.

What would you do?

Reply:
> Basically a version of let that carries an "primary SUT" connotation?

They are a `let` under the hood ;) Yes I would, but I don't even use `subject` in my own tests.

Reply:
Okay, I've redone all the specs in this style, and I agree that it is an improvement in readability! Thank you for helping me level up my RSpec skills!